cmd/git-sync/main.go

```diff
@@ -155,6 +159,93 @@ const (
 	submodulesOff       = "off"
 )
 
+const (
+	errorFile = "error.json"
+)
+
+type customLogger struct {
+	logr.Logger
+	errorFile string
+}
+
+func (l customLogger) Error(err error, msg string, kvList ...interface{}) {
+	l.Logger.Error(err, msg, kvList...)
+	if !*flExportError {
+		return
+	}
+	payload := struct {
+		Msg string
+		Err string
+		Args map[string]interface{}
+	} {
+		Msg: msg,
+		Err: err.Error(),
+		Args: map[string]interface{}{},
+	}
+	for i := 0; i < len(kvList); i += 2 {
+		k, ok := kvList[i].(string)
+		if !ok {
+			jb, _ := json.Marshal(kvList[i])
+			fmt.Fprintf(os.Stderr,"key is not a string: %s\n", string(jb))
+			os.Exit(1)
+		}
+		var v interface{}
+		if i+1 < len(kvList) {
+			v = kvList[i+1]
+		}
+		payload.Args[k] = v
+	}
+	jb, err := json.Marshal(payload)
+	if err != nil {
+		fmt.Fprintf(os.Stderr,"ERROR: can't encode payload: %v\n", err)
+		os.Exit(1)
+	}
+	l.writeContent(jb)
+}
+
+// exportError exports the error to the error file if --export-error is enabled.
+func (l *customLogger) exportError(content string) {
+	if !*flExportError {
+		return
+	}
+	l.writeContent([]byte(content))
+}
+
+// writeContent writes the error content to the error file.
+func (l *customLogger) writeContent(content []byte) {
+	tmpFile, err := ioutil.TempFile(*flRoot, "tmp-err-")
```

Comment:
don't read global flags - stick the root into the logger instance at creation.

Reply:
DONE